mailsync

```diff
@@ -1,5 +1,84 @@
 #!/bin/bash
-migsync etc/${1%/} ~/etc
-migsync etc/${1%/} ~/etc
-migsync mail/${1%/} ~/mail
-migsync mail/${1%/} ~/mail
\ No newline at end of file
+
+# check to see if there's an argument
+
+if [ -z "$1" ]; then
+  echo -e 'USAGE: mailsync [domain]'
+  exit 1
+fi 
+
+if [ ! -d ~/migration/migstuff/ ] ; then  
+  mkdir ~/migration/migstuff/ 2>/dev/null
+  mkdir ~/migration/migstuff/logs/ 2>/dev/null
+elif [ ! -d ~/migration/migstuff/logs/ ] ; then 
+  mkdir ~/migration/migstuff/logs/ 2>/dev/null
+  mkdir ~/migration/migstuff/logs/"$1"/ 2>/dev/null
+fi
+
+echo ''
+timestamp=$(date +%s)
+
+# function to count inodes in mail dir for domain 
+
+echo ' ' > ~/migration/migstuff/logs/"$1"/beforesync_"$1"_"$timestamp".txt
+echo ' ' > ~/migration/migstuff/logs/"$1"/aftersync_"$1"_"$timestamp".txt
+
+function mail_Count() {
+  (cd ~/mail/"$1" ; find . -printf "%h\n" | cut -d/ -f-2 | sort | uniq -c | sort -rn)
+}
+
+# function to format and send count to file 
+
+function count_and_Format() {
+  if [[ -d ~/mail/$1 ]]; then 
+    mail_Count $1 | sed -rn "s/\s*\.\/(.*+)/\t\1@$1/p"
+  fi 
+}
+
+echo -e "Migrating mail for $1\n"
+
+# record mail box state from before sync
+
+echo -e 'PREVIOUS' 1> ~/migration/migstuff/logs/"$1"/beforesync_"$1"_"$timestamp".txt
+count_and_Format $1 1>> ~/migration/migstuff/logs/"$1"/beforesync_"$1"_"$timestamp".txt 
+
+# do the actual sync 
+
+sourcecreds=$(cat ~/migration/migstuff/source.txt)
+
+#check rsync version for progress option
+rsync_version=$(rsync --version | grep -oP 'rsync\s+version\s\K(\d\.\d)' | sed -rn 's/\.//p')
+
+if [[ "$rsync_version" -lt "31" ]]; then 
+  echo "Syncing mail, no progress bar available"
+  # sync etc
+  rsync --dry-run --ignore-existing -avP -e "ssh -i ~/migration/migstuff/id_rsa" $sourcecreds:~/etc/${1%/} ~/etc 1> ~/migration/migstuff/logs/"$1"/dryrun_mailsync_etc_log_"$timestamp".txt &
```

Comment:
line 55 and 62 look identicle. the code does not need to be in both sides of the if statement. can be moved above

Reply:
consolidated 2 lines from that part, had several that were the same and put them in front